sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "opentelemetry/version.h"
+#include "opentelemetry/common/attribute_value.h"
+
+#include <memory>
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+class Sampler
+{
+public:
+  /**
+   * Initialize a new tracer.
+   * @param processor The span processor for this tracer. This must not be a
+   * nullptr.
+   */
+  virtual Decision shouldSample();
+
+  /**
+   * Returns the description of this {@code Sampler}. This may be displayed on debug pages or in the
+   * logs.
+   *
+   * <p>Example: "ProbabilitySampler{0.000100}"
+   *
+   * @return the description of this {@code Sampler}.
+   */
+  virtual std::string getDescription();
+};
+
+class Decision
+{
+public:
+  /**
+   * Return sampling decision whether span should be sampled or not.
+   *
+   * @return sampling decision.
+   */
+  virtual bool isSampled();
```

Comment:
I would call this method `GetDecision` (the class name being `SamplingResult`). It could return a `Decision` enum (either `NOT_RECORD`, `RECORD` or `RECORD_AND_SAMPLE`, according to the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample).